crates/wit-component/src/encoding.rs

```diff
@@ -2418,6 +2503,32 @@ impl<'a> Shims<'a> {
     }
 }
 
+fn task_return_options_and_type(
+    resolve: &Resolve,
+    ty: Option<Type>,
+    async_: bool,
```

Comment:
When would this be `true`?

Reply:
Ah this is a holdover from prior to the clarification of https://github.com/WebAssembly/component-model/issues/453, I'll remove it